ui/components/app/qr-hardware-state-watcher/qr-hardware-state-watcher.js

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getCurrentQRHardwareState } from '../../../selectors';
+import Popover from '../../ui/popover';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { cancelReadQRHardwareCryptoHDKey as cancelReadQRHardwareCryptoHDKeyAction } from '../../../store/actions';
+import QRHardwareWalletImporter from './qr-hardware-wallet-importer';
+
+const QRHardwareWalletWatcher = () => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const walletImporterCancel = useCallback(
+    () => dispatch(cancelReadQRHardwareCryptoHDKeyAction()),
+    [dispatch],
+  );
+
+  const [showWalletImporter, setShowWalletImporter] = useState(false);
+  const showPopover = useMemo(() => showWalletImporter, [showWalletImporter]);
+
+  const qrHardware = useSelector(getCurrentQRHardwareState);
+  const { sync } = qrHardware;
+
+  useEffect(() => {
+    setShowWalletImporter(sync.reading);
+  }, [sync.reading]);
+
+  const title = useMemo(() => {
+    let _title = '';
+    if (showWalletImporter) {
+      _title = t('QRHardwareWalletImporterTitle');
+    }
+    return _title;
```

Comment:
Can this just be `return showWalletImporter ? t('QRHardwareWalletImporterTitle') : ''`?

Reply:
That's because we will introduce sign request component in next PR which will be embedded in qr-hardware-popover. But it's fine to modify the code in this PR.